pkg/frontend/tgt/frontend.go

```diff
@@ -138,3 +150,28 @@ func (t *Tgt) Expand(size int64) error {
 	}
 	return nil
 }
+
+// attemptUnfreezeFilesystem attempts to identify a mountPoint for the Longhorn device and unfreeze it. Under normal
+// conditions, it will not find a filesystem, and if it finds a filesystem, it will not be frozen.
+// attemptUnfreezeFilesystem is only relevant for volumes run with a tgt-blockdev frontend, as only these volumes have a
+// Longhorn device on the node to format and mount.
+func (t *Tgt) attemptUnfreezeFilesystem() {
+	// We do not need to switch to the host mount namespace to get mount points here. Usually, longhorn-engine runs in a
+	// container that has / bind mounted to /host with at least HostToContainer (rslave) propagation.
+	// - If it does not, we likely can't do a namespace swap anyway, since we don't have access to /host/proc.
+	// - If it does, we just need to know where in the container we can access the mount points to unfreeze.
+	if t.frontendName == types.EngineFrontendBlockDev {
+		mounter := mount.New("")
+		mountPoints, err := mounter.List()
+		if err != nil {
+			logrus.WithError(err).Warn("Failed to list mount points while deciding whether or not unfreeze")
+		}
+		endpoint := t.Endpoint()
+		for _, mountPoint := range mountPoints {
+			if mountPoint.Device == endpoint {
+				util.AttemptUnfreezeFilesystem(mountPoint.Path, lhexec.NewExecutor(), false, logrus.New())
+				break
```

Comment:
Should we prioritize the mount point we created first (the one at `freezePointDirectory  = "/var/lib/longhorn/freeze"`)?

Reply:
It probably wouldn't hurt. I'll add this functionality in an upcoming pass.